order-book.js

```diff
@@ -0,0 +1,99 @@
+const reconcileOrder = (existingBook, incomingOrder) => {
+
+  switch (existingBook.length) {
+    case 0:
+      return toBook(existingBook, incomingOrder)
+
+    default:
+      return determineOrder(existingBook, incomingOrder)
+  }
+
+}
+
+const determineOrder = (existing, incoming) => {
+
+  for (let i = 0; i < existing.length; i++) {
+
+    switch (incoming.type) {
+      case existing[i].type:
+        return toBook(existing, incoming)
+
+      default:
+        return selectOrder(existing, incoming, i)
+    }
+  }
+}
+
+const toBook = (existing, incoming) => {
+  existing.push(incoming)
+  return existing
+}
+
+const selectOrder = (existing, incoming, index) => {
+  switch (incoming.price) {
+    case existing[index].price:
+      return resolveOrder(existing, incoming, index)
+    default:
+      return toBook(existing, incoming)
+  }
+}
+
+const resolveOrder = (existing, incoming, index) => {
+  switch (true) {
```

Comment:
You would never write a switch statement like this. Switches are for looking at a single variable and running different operations for each possible value of that variable. If you don't have a single variable with a set of 3 or 4 possible values, you don't use a switch.